compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -7048,4 +7050,17 @@ private enum BasicTypes {
         ANYDATA,
         JSON
     }
+
+    /**
+     * Holds common analyzer data between {@link TypeChecker} and {@link SemanticAnalyzer}.
+     */
+    public static class CommonAnalyzerData {
```

Comment:
Shall we rename this to `QueryAnalyzerData`, because this is specific to query?

Reply:
On a second thought, can we maintain two separate `QueryAnalyzerData` static inner classes in AnalyzerData to exchange the data between `TypeChecker` and `SemanticAnalyzer`, instead of having this common class in `Types`?